File: src/android/com/example/fibonacci/FibonacciPlugin.java
package com.example.fibonacci;

import org.apache.cordova.CallbackContext;
import org.apache.cordova.CordovaInterface;
import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CordovaWebView;
import org.json.JSONArray;
import org.json.JSONException;

public class FibonacciPlugin extends CordovaPlugin {

  private FibonacciPluginKt plugin = new FibonacciPluginKt();

  @Override
  public void initialize(CordovaInterface cordova, CordovaWebView webView) {
    super.initialize(cordova, webView);
    plugin.initialize(cordova, webView);
  }

  @Override
  public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws
    JSONException {
    return plugin.execute(action, args, callbackContext);
  }
}

File: src/android/com/example/fibonacci/FibonacciPlugin.kt
package com.example.fibonacci

import android.util.Log
import org.apache.cordova.*
import org.json.JSONArray
import java.util.*

class FibonacciPlugin : CordovaPlugin() {

  val TAG: String = "FibonacciPlugin";

  override fun initialize(cordova: CordovaInterface?, webView: CordovaWebView?) {
    super.initialize(cordova, webView)

    Log.d(TAG, "Initializing FibonacciPlugin in Kotlin")
  }

  private fun echo(phrase: String?) {
    Log.d(TAG, phrase);
  }

  private fun getDate(callbackContext: CallbackContext?) {
    val result = PluginResult(PluginResult.Status.OK, Date().toString())
    callbackContext?.sendPluginResult(result);
  }

  override fun execute(action: String?, args: JSONArray?, callbackContext: CallbackContext?): Boolean {
    when (action) {
      "echo" -> this.echo(args?.getString(0))
      "getDate" -> this.getDate(callbackContext)
      else -> return false;
    }
    return true;
  }
}
